packages/builder/src/types/routing.ts

```diff
@@ -0,0 +1,174 @@
+import { ComponentType } from "svelte"
+import { SvelteComponent } from "svelte"
+
+export interface Routing {
+  params?: Record<string, any>
+}
+
+export const enum Target {
+  Self = "_self",
+  Blank = "_blank",
+}
+
+export interface RouteHREF {
+  url: string
+  target: Target
+}
+
+export const isRouteHREF = (href: string | RouteHREF): href is RouteHREF => {
+  return typeof href === "object" && href !== null && "url" in href
+}
+
+export const isSettingIcon = (
+  icon: string | RouteIcon | undefined
+): icon is RouteIcon => {
+  return (
+    typeof icon === "object" &&
+    icon !== null &&
```

Comment:
To handle undefineds
```suggestion
    icon &&
```